### PR TITLE
Add line info for markdown parser.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

**Comment by vwxyzh** [comment]
for commit 1:
1. only work for markdownlite, and its test.
2. some tokens line number is incorrect (e.g. tokens in lists or tables).

**Comment by vwxyzh** [comment]
@chenkennt @superyyrrzz @fenxu

File: src/Microsoft.DocAsCode.MarkdownLite/IHasLineInfo.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.MarkdownLite
+{
+    public interface IHasLineInfo
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/IHasLineInfo.cs]: It isn't used for now. Does this needed?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.MarkdownLite/IHasLineInfo.cs]: removed.

File: src/Microsoft.DocAsCode.MarkdownLite/MarkdownParserContext.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/MarkdownParserContext.cs]: move `using` into namespace

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.MarkdownLite/MarkdownParserContext.cs]: fixed.

File: src/Microsoft.DocAsCode.MarkdownLite/LineInfo.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            LineNumber = lineNumber;
+        }
+
+        public string File { get; }
+
+        public int LineNumber { get; }
+
+        public bool HasInfo => File != null && LineNumber != 0;
+
+        public LineInfo Move(int offset)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.MarkdownLite/LineInfo.cs]: Minor: `Move` => `Offset`? `Move` makes me think there's some property changed after calling `Move`...

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request closed without merging**